finger/views.py

```diff
@@ -29,6 +30,54 @@ def upload_json(request):
         'form': form,
     })
 
+@login_required
+def mail_members(request):
+    if not (request.user.is_staff and request.user.is_superuser):
+        return HttpResponseNotAllowed()
+
+    if request.method == 'POST':
+        form = MailMembersForm(request.POST)
+    else:
+        form = MailMembersForm()
+
+    if form.is_valid():
+        members = form.get_members()
+
+        subject = form.cleaned_data['subject']
+        message = form.cleaned_data['message']
+        sender = request.user.username + '@stacken.kth.se'
+
+        recipients = form.cleaned_data['recipients']
+        if recipients:
+            recipients = [("", recipient) for recipient in recipients.split(',')]
+        else:
+            # todo send to all email addresses
+            recipients = [(f"{member.first_name} {member.last_name}", member.email) for member in members]
+
+        cc = form.cleaned_data['cc']
+        if not cc:
+            cc = "styrelsen@stacken.kth.se"
+
+        context = {
+                'recipients' : recipients,
+                'form' : form,
+                'test' : True,
+        }   
+
+        if request.POST.get("do_it", "") == "true":
+            context['test'] = False
+            to = [email for (_, email) in recipients]
+            to.append(cc)
+
+            send_mail(subject, message, sender, recipients)
```

Comment:
This is not correct, this will add all recipient emails to the TO header, **exposing all addresses** to all members. A possible solution would to add them to BCC but that will probably not scale... in my opinion make this a loop like `send.py`, or build a list and send it with `send_mass_mail(...)`

Reply:
Håller med. Är nu fixat.